### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by govardhanreddyc79**
Wishlist Management implemented.
- Add to wishlist in product listing
- Move to wishlist from cart
- Go to wishlist from cart
- Move to cart from wishlist
- Clear wishlist

image

### CODE REVIEW

File: src/context/wishlist-context.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const WishlistContext = createContext();
+
+const useWishlist = () => useContext(WishlistContext);
+
+const WishlistProvider = ({children}) => {
+
+const [wishlistState, wishlistDispatch ] = useReducer(wishlistReducer, {
+        wishlistProducts : [],
+        wishlistProductsCount:0
```
Comment (arpit01923) [inline src/context/wishlist-context.js:12]: I think there is no need of this, just an unnecessary code. To get length directly use wishlistProduct.length method

File: src/reducer/wishlistReducer.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export const wishlistReducer = ( state = initialState, {type, payload}) => {
+    
+    switch(type){
+        case "MOVE_TO_WISHLIST":
+            console.log(state.wishlistProducts)
+            return{
+                ...state,
+                wishlistProducts:
+                        [...state.wishlistProducts, payload],
+                wishlistProductsCount: state.wishlistProductsCount + 1,
```
Comment (arpit01923) [inline src/reducer/wishlistReducer.js:13]: unnecessary code

File: src/reducer/wishlistReducer.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            }
+        case "REMOVE_FROM_WISHLIST":
+            return {
+                ...state,
+                wishlistProducts:
+                [...state.wishlistProducts.filter(item => item._id !== payload._id)],
+                wishlistProductsCount: state.wishlistProductsCount - 1
```
Comment (arpit01923) [inline src/reducer/wishlistReducer.js:20]: unnecessary code

**Review by arpit01923 — APPROVED** [review]
- clean UI
- good folder structure
- left suggestions, maybe useful

File: src/reducer/wishlistReducer.js (hunk 3 above)
Comment (govardhanreddyc79) [reply src/reducer/wishlistReducer.js:20]: okay

File: src/components/cartProductCard.jsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
 export const CartProductCard = ({cartProduct}) => {
-    console.log(cartProduct.quantity)
-    const {productName, imageSource, imageAlt, price, quantity } = cartProduct;
+    const {productName, imageSource, imageAlt, price, quantity, _id} = cartProduct;
 
     const { cartDispatch } = useCart();
-    
+    const { wishlistState, wishlistDispatch } = useWishlist();
+    const { wishlistProducts, wishlistProductsCount } = wishlistState;
+    console.log(wishlistProducts, wishlistProductsCount)
```
Comment (SanketDhabarde) [inline src/components/cartProductCard.jsx:10]: Remove logs

**Review by SanketDhabarde — APPROVED** [review]
Looks good to me
- Like the use of proper names for state and dispatch
- Good UI
- Clean and easy to understand code

**Pull request merged**